src/Portable.Xaml/Portable.Xaml.ComponentModel/DateTimeConverter.cs

```diff
@@ -45,8 +58,28 @@ public override object ConvertTo (ITypeDescriptorContext context, CultureInfo cu
 
                 return date.ToString(format, culture);
             }
-			return base.ConvertTo (context, culture, value, destinationType);
-		}
-	}
+            return base.ConvertTo(context, culture, value, destinationType);
+        }
+
+        private string RemoveUnnecessaryZeros(string dateFormatted, DateTimeKind kind)
+        {
+            string conciseDate = dateFormatted.Trim();
+            if (kind == DateTimeKind.Utc)
+            {
+                if (UtcTidyUpPattern.IsMatch(dateFormatted))
+                    conciseDate = UtcTidyUpPattern.Replace(dateFormatted, "Z");
```

Comment:
no need for Match + Replace, as it will then parse the regex twice.  Just simply doing the Replace will work here (and for the LocalTidyUpPatterns).